src/client.ts

```diff
@@ -5188,6 +5293,127 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         return this.sendMessage(roomId, threadId, content);
     }
 
+    /**
+     * Send a future for a timeline event.
+     *
+     * Note: This endpoint is unstable, and can throw an `Error`.
+     *   Check progress on [MSC4140](https://github.com/matrix-org/matrix-spec-proposals/pull/4140) for more details.
+     */
+    // eslint-disable-next-line
+    public _unstable_sendFuture<K extends keyof TimelineEvents, F extends ISendFutureRequestOpts>(
+        roomId: string,
+        futureOpts: F,
+        eventType: K,
+        content: TimelineEvents[K],
+        txnId?: string,
+    ): Promise<ISendFutureResponse<F>>;
+    // eslint-disable-next-line
+    public _unstable_sendFuture<K extends keyof TimelineEvents, F extends ISendFutureRequestOpts>(
+        roomId: string,
+        futureOpts: F,
+        threadId: string | null,
+        eventType: K,
+        content: TimelineEvents[K],
+        txnId?: string,
+    ): Promise<ISendFutureResponse<F>>;
+    // eslint-disable-next-line
+    public async _unstable_sendFuture<F extends ISendFutureRequestOpts>(
+        roomId: string,
+        futureOpts: F,
+        threadIdOrEventType: string | null,
+        eventTypeOrContent: string | IContent,
+        contentOrTxnId?: IContent | string,
+        txnIdOrVoid?: string,
+    ): Promise<ISendFutureResponse<F>> {
```

Comment:
Could we please avoid the `threadId` overload, just have the caller specify `null` for "Main" timeline